server/broadcast.go

```diff
@@ -276,9 +276,20 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) error {
 
 	for {
 		// if fails, retry; rudimentary
-		if err := transcodeSegment(cxn, seg, name); err == nil {
+		err := transcodeSegment(cxn, seg, name)
+		// no error, transcode succeeded
+		if err == nil {
 			return nil
 		}
+
+		// non retryable error, close stream
+		if shouldStopStream(err) {
```

Comment:
The updated impl of `shouldStopStream()` would close the RTMP conn when there are no sessions available for a segment which might not be desirable. I think there is a difference between a non-retryable error that closes the RTMP conn vs. a non-retryable error that allows the stream to continue (but drops the current segment). We currently skip the segment when there are no sessions available, but do not close the RTMP conn.